JOSESwift/Sources/CryptoImplementation/EC.swift

```diff
@@ -95,33 +95,46 @@ public enum ECCurveType: String, Codable {
     }
 }
 
-fileprivate extension SignatureAlgorithm {
-    typealias DigestFunction = (
-            UnsafeRawPointer?,
-            UInt32,
-            UnsafeMutablePointer<UInt8>?
-    ) -> UnsafeMutablePointer<UInt8>?
-
-    func createDigest(input: Data) throws -> [UInt8] {
-        guard
-                let computedDigestLength = digestLength,
-                let computedDigestFunction = digestFunction
-                else {
-            throw ECError.invalidCurveDigestAlgorithm
+// converting integers to and from DER encoded ASN.1 as described here:
+// https://docs.microsoft.com/en-us/windows/desktop/seccertenroll/about-integer
+fileprivate extension Data {
+    func withLengthFixedTo(_ octetLength: Int) -> Data {
+        let varLength = self.count
+        if varLength > octetLength + 1 {
+            fatalError("Unable to parse ASN.1 Integer")
+        }
+        if varLength == octetLength + 1 {
+            return self.dropFirst()
         }
-        var digest = [UInt8](repeating: 0, count: computedDigestLength)
-        _ = computedDigestFunction(Array(input), UInt32(input.count), &digest)
-        return digest
+        if varLength == octetLength {
+            return self
+        }
+        if varLength < octetLength {
+            return Data.init(count: octetLength - varLength) + self
```

Comment:
About the documentation of the padding:

As far as I can tell the [JWS standard only says](https://tools.ietf.org/html/rfc7515#appendix-A.3.1):
> the R and S values, given as octet sequences representing big-endian integers

and later

> The JWS Signature is the value R || S

R and S need to have a fixed length, because the length of R and S is not encoded anywhere else explicitly so otherwise there is no way for the verifying party to tell where R ends and S begins.

I cannot find this mentioned more explicitly in the standard.